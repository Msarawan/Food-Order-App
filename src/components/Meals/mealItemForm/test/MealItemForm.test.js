import MealItemForm from "../../mealItemForm/MealItemForm";
import React from "react";
import {mount} from "enzyme";


jest.mock('../../../UI/input/Input');

describe("Food Order app testing", () => {
  
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<MealItemForm onSubmit='{submitHandler}' onClick='{addToCartHandler}' />);
  });

  test("renders correctly", () =>{
    wrapper=mount(<MealItemForm/>)
  })

  test("render the Add button", () => {
     expect(wrapper.find('button').text()).toContain('+ Add')})

  test("Add the items by clicking the add button", () => {
    expect(wrapper.props().onClick).toEqual('{addToCartHandler}');
      console.log(wrapper.find('MealItemForm'))
   });

  test("render sumbit handler", () => {
    expect(wrapper.props().onSubmit).toEqual('{submitHandler}');
  })

 test('does not reload page after submition', () => { 
  jest.spyOn(React, 'useRef').mockReturnValueOnce({current:{value:'2'}});
  const props = {onAddToCart:jest.fn()}
  wrapper = mount(<MealItemForm {...props}/>);
  expect(wrapper).toMatchSnapshot();


 const event = { preventDefault:() => {} }
  jest.spyOn(event, 'preventDefault')
  wrapper.find('form').simulate('submit', event)
  expect(event.preventDefault).toBeCalled()
  })

})




