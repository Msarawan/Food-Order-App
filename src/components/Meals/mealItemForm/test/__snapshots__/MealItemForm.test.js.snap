// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Order app testing does not reload page after submition 1`] = `
<MealItemForm
  onAddToCart={[MockFunction]}
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <MockedInput
      input={
        {
          "defaultValue": "1",
          "id": "amount_undefined",
          "max": "5",
          "min": "1",
          "step": "1",
          "type": "number",
        }
      }
      label="Amount"
    />
    <button>
      + Add
    </button>
  </form>
</MealItemForm>
`;
